### PR TITLE
Add test for purged backup files from remote storage

### PR DESCRIPTION
**Pull request by ivanmp91**

### CODE REVIEW

File: tests/integration/features/steps/integration_steps.py [hunk 1]
```diff
@@ -638,6 +638,36 @@ def _i_cannot_see_the_backup_named_backupname_when_i_list_the_backups(
...
+    context, table_name, keyspace
+):
+    storage = Storage(config=context.medusa_config.storage)
+    path = os.path.join(
+        storage.prefix_path + context.medusa_config.storage.fqdn, "data", keyspace, table_name
+    )
+    sb_files = len(storage.storage_driver.list_objects(path))
+
+    node_backups = storage.list_node_backups()
+    # Parse its manifest
```
Comment (adejanovski) [inline tests/integration/features/steps/integration_steps.py:652]: It's not what you're doing at this line. This comment belongs to line 656 instead.

**Review by adejanovski — CHANGES_REQUESTED** [review]
I added some logging and you still have differences indeed. 

This is what's in the remaining manifests:
```
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-TOC.txt
```

And this is what's in the data directory:
```
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-1-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-3-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-4-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-5-big-TOC.txt
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-CompressionInfo.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Data.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Digest.adler32
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Filter.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Index.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Statistics.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-Summary.db
test-e3ac40f06aaf11eb80045d45ffed10f6/lb-6-big-TOC.txt
```

You can see there that the `lb-1` sstable is missing from the remaining manifests, but still exists in the backup `data` directory. The purge didn't fully remove the orphaned files then.
The test is good, it's the purge code that is faulty here it seems.

File: tests/integration/features/steps/integration_steps.py (hunk 1 above)
Comment (ivanmp91) [reply tests/integration/features/steps/integration_steps.py]: right! This should be fixed now

**Comment by ivanmp91** [comment]
Awesome! Thanks for checking it out @adejanovski ! is there anything else that it should be addressed here? or are we good to go merging the latest changes here with that PR? https://github.com/thelastpickle/cassandra-medusa/pull/276

**Comment by adejanovski** [comment]
@ivanmp91, we can't merge the PR if it comes with failing tests. The test showed that there's something broken in the purge process which needs fixing.
I'll see if I can find what's wrong in the purge code and report back.

**Comment by ivanmp91** [comment]
@adejanovski I think the new test is failing because it doesn't include the storage prefix patch of the other PR. The other PR already includes this test (https://github.com/thelastpickle/cassandra-medusa/pull/276/commits/5c98c167a02df924e997e15601939a6dddaf1b81) and the test is passing there because of that patch. I created this branch in top of master just to see if the test was failing or not and compare with the result with the other branch.

**Pull request closed without merging**